### PR TITLE
[test][gardening] Prefer `os(macOS)` over `os(OSX)`

### PR DESCRIPTION
**Pull request by ikesyo**
Refs: #15106, #15087, #13315

### CODE REVIEW

**Comment by ikesyo** [comment]
@swift-ci smoke test

**Review by compnerd — APPROVED** [review]
LGTM

File: test/Parse/ConditionalCompilation/basicIdentity.swift [hunk 1]
```diff
@@ -27,7 +27,7 @@ var i = Foo()
 #endif
 var j = i
 
-#if os(OSX)
+#if os(macOS)
```
Comment (CodaFi) [inline test/Parse/ConditionalCompilation/basicIdentity.swift]: Be careful changing the ConditionalCompilation tests.  I would like this one preserved and a `macOS` test added here.

Comment (ikesyo) [reply test/Parse/ConditionalCompilation/basicIdentity.swift]: Good point! I think the both `OSX` and `macOS` conditions are specifically tested in https://github.com/apple/swift/blob/c26f40b45b3d21fb5844c71549ee23baa4f7b855/test/Parse/ConditionalCompilation/x64OSXTarget.swift#L4-L15 (https://github.com/apple/swift/pull/3395/files#diff-ddecd035914171df461e408178c90007),

but preserving the `OSX` test and adding a `macOS` test to this file would be good anyway.

**Comment by ikesyo** [comment]
@swift-ci Please smoke test and merge

**Comment by ikesyo** [comment]
@swift-ci smoke test and merge

**Pull request merged**